src/protocol/context/malicious.rs

```diff
@@ -0,0 +1,101 @@
+use crate::ff::Field;
+use crate::helpers::messaging::{Gateway, Mesh};
+use crate::helpers::Role;
+use crate::protocol::context::{ContextInner, ProtocolContext, SemiHonestProtocolContext};
+use crate::protocol::malicious::SecurityValidatorAccumulator;
+use crate::protocol::prss::{
+    Endpoint as PrssEndpoint, IndexedSharedRandomness, SequentialSharedRandomness,
+};
+use crate::protocol::{Step, Substep};
+use crate::secret_sharing::{MaliciousReplicated, Replicated};
+use std::sync::Arc;
+
+/// Represents protocol context in malicious setting, i.e. secure against one active adversary
+/// in 3 party MPC ring.
+#[derive(Clone, Debug)]
+pub struct MaliciousProtocolContext<'a, F: Field> {
+    inner: ContextInner<'a>,
+    accumulator: SecurityValidatorAccumulator<F>,
+    r_share: Replicated<F>,
```

Comment:
It seems like the inner object will need to be per-implementation rather than shared.  Otherwise, we're duplicating `r_share` and `accumulator` when the context is cloned.